src/Content/Text/BBCode.php

```diff
@@ -2112,6 +2116,31 @@ function ($matches) {
 		return trim($text);
 	}
 
+	private static function shortenLinkDescription(string $text): string
```

Comment:
Why not do this when converting `[url]` tags to HTML instead of using a transitional form that mixes BBCode and HTML entities for square brackets?

Reply:
I don't like to have got multiple places where a link gets converted into HTML to make sure that it is done in just a single way.

Reply:
We're already in `BBCode::convert`, so it would still be in a single place?

Reply:
There would still be multiple lines in that places.

Reply:
There's already a `BBCode::convertUrlForActivityPubCallback` that does the `[url]` tag to shortened-looking `<a>` tag, it's just a matter of using it no matter the `$simple_html` value.

Reply:
Later in the code we translate the `url` element differently depending if it is linked to the same or a remote system. If we would directly create HTML by using that function, we had to add this check for a local link there as well.

Concerning the entity conversion in the BBCode links: We already had to do so at different places. It is no technical problem and the HTML output stays rhe same, see the test results. I now applied some more changes in the parser for the link handling.